### PR TITLE
Add support for different env

### LINKED ISSUE #434: Improve rollup build config for development, staging_production, and staging/goerli
### Problems
1. Right now rollup build set to development will default to goerli.
1. Right now rollup build in staging needs to be hacked to point towards production.
Update rollup build so we don't have to hack
1. Right now staging/goerli gives you APY > 10,000%

### DoD:
- [x] (1) Development build works correctly against ganache
- [x] (1) Development build should default to 20 OCEAN / week.
- [x] (2) Staging-Production build works without having to hack code.
- [x] (3) Goerli should not reward you with 10,000% APY. Default to 20 OCEAN / week.

### PR DESCRIPTION
**Pull request by KatunaNorbert**
Fixes #434.

### CODE REVIEW

**Review by EnzoVezzaro — DISMISSED** [review]
LGTM

**Comment by KatunaNorbert** [comment]
Look like we need another approval to be able to merge this one

**Comment by EnzoVezzaro** [comment]
> Looks like we need another approval to be able to merge this one

@jamiehewitt15 can you please quickly review this

File: .env.example [hunk 1]
```diff
@@ -1,3 +1,30 @@
...
+# You can enable or disable rollup livereload to start a server and reload the page on saving changes.
+# This is usualy enabled on local and disable on production.
+START_RELOAD_SERVER = true
+
+
+
+
+# You can also set your custom environment variables by adding a different value for NODE_ENV then the specified ones above, and manually setting the other variables.
+
+# set one of Ocean subgraphs from the available subgrapsh list from https://github.com/oceanprotocol/ocean-subgraph.
```
Comment (jamiehewitt15) [inline .env.example:25]: We should keep the examples in the comments like we have currently. That makes it easier for people to choose which subgraph they want:
```suggestion

# choose one of the following local subgraph (dev), goerli (staging), and mainnet (prod)
# Dev: http://172.15.0.15:8000/subgraphs/name/oceanprotocol/ocean-subgraph
# Staging: https://v4.subgraph.goerli.oceanprotocol.com/subgraphs/name/oceanprotocol/ocean-subgraph
# Production: https://v4.subgraph.mainnet.oceanprotocol.com/subgraphs/name/oceanprotocol/ocean-subgraph
```

File: .env.example [hunk 2]
```diff
@@ -1,3 +1,30 @@
+# Choose one of the following environment configuration: local (development), goerli (staging), mainnet (production)
+# This will automatically set other required environment variables to point to choosen configuration.
+
+# Development: point to your local ganache chain. Usit it when running Ocean stack locally with Barge.
+NODE_ENV = "development"
```
Comment (jamiehewitt15) [inline .env.example:5]: I'd suggest that we should comment out two of these values for `NODE_ENV`, that way this file can work with the default value as soon as it's copied without the user having to come in here and commenting things out

**Comment by jamiehewitt15** [comment]
Maybe I've done something wrong with the setup. When I try and run this locally with barge, I get this message saying switch to Goerli:

image

image

File: .env.example (hunk 1 above)
Comment (KatunaNorbert) [reply .env.example:25]: The ideal would be for user to set one of the mentions for `NODE_ENV` and then they won't need to configure anything.
If they want something else there is an example at the end of the file with all the variables he needs to set. The subgraphs are not listed but we give one example and point user to all our subgraphs so he can choose one from there.

Comment (KatunaNorbert) [reply .env.example:25]: I will comment not necessary configs

Comment (jamiehewitt15) [reply .env.example:25]: It doesn't say anywhere in the subgraph readme the address of the subgraph in barge

**Comment by KatunaNorbert** [comment]
@jamiehewitt15 I'm not sure, works fine for me. Have rerun the app after you changed the env variable? If not, you need to, otherwise the old one remains saved in memory.

**Comment by jamiehewitt15** [comment]
> @jamiehewitt15 I'm not sure, works fine for me. Have you rerun the app after you changed the env variable? If not, you need to, otherwise the old one remains saved in memory.

Yeah, I've stopped it and started it again. Are the values I have in the `.env` the same as yours?

File: .env.example (hunk 1 above)
Comment (KatunaNorbert) [reply .env.example:25]: That's right, I just checked, looks like subgraphs readme should be updated

File: .env.example (hunk 2 above)
Comment (KatunaNorbert) [reply .env.example]: done

**Comment by KatunaNorbert** [comment]
Yes, If I use your exactly it shows me a dropdown to select network, because supported networks list is set to have multiple values. If I comment that It points me to development

**Comment by EnzoVezzaro** [comment]
@KatunaNorbert I think those "//" meant to be "#", right?

the final .env.example should be?

`
# Choose one of the following environment configuration: local (development), goerli (staging), mainnet (production)
# This will automatically set other required environment variables to point to choosen configuration.

# Development: point to your local ganache chain. Usit it when running Ocean stack locally with Barge.
# NODE_ENV = "development"

# Staging: point to Goerli network.
NODE_ENV = "staging"

# Production: point to Ethereum Mainnet.
# NODE_ENV = "production"

# You can enable or disable rollup livereload to start a server and reload the page on saving changes.
# This is usualy enabled on local and disable on production.
START_RELOAD_SERVER = true

# You can also set your custom environment variables by adding a different value for NODE_ENV then the specified ones above, and manually setting the other variables.

# set one of Ocean subgraphs from the available subgrapsh list from https://github.com/oceanprotocol/ocean-subgraph.
# SUBGRAPH_API = "https://v4.subgraph.goerli.oceanprotocol.com/subgraphs/name/oceanprotocol/ocean-subgraph"

# chain where ve contracts are deployed - 8996 is development
# VE_SUPPORTED_CHAINID = 8996

# list of suppported chainIDs (used for filtering asset only)
# SUPPORTED_CHAIN_IDS = [8996]

# api endpoint to get the pools list
# BACKEND_API = "https://test-df-sql.oceandao.org"

# your infura key
# INFURA_KEY = "01SAMPLE9234KEY"

# set to "enabled" if you want to enable wallet address debugging inside app
# DEBUGGING = "enabled"
`

on my end this is not working. The app crashes as soon as loads. 


____ dev is working _____

For me this is working:

# Choose one of the following environment configuration: local (development), goerli (staging), mainnet (production)
# This will automatically set other required environment variables to point to choosen configuration.

# Development: point to your local ganache chain. Usit it when running Ocean stack locally with Barge.
NODE_ENV = "development"

# Staging: point to Goerli network.
# NODE_ENV = "staging"

# Production: point to Ethereum Mainnet.
# NODE_ENV = "production"

# You can enable or disable rollup livereload to start a server and reload the page on saving changes.
# This is usualy enabled on local and disable on production.
START_RELOAD_SERVER = true

# You can also set your custom environment variables by adding a different value for NODE_ENV then the specified ones above, and manually setting the other variables.

# set one of Ocean subgraphs from the available subgrapsh list from https://github.com/oceanprotocol/ocean-subgraph.
SUBGRAPH_API = "http://127.0.0.1:9000/subgraphs/name/oceanprotocol/ocean-subgraph"

# chain where ve contracts are deployed - 8996 is development
VE_SUPPORTED_CHAINID = 8996

# list of suppported chainIDs (used for filtering asset only)
SUPPORTED_CHAIN_IDS = [8996]

# api endpoint to get the pools list
BACKEND_API = "https://test-df-sql.oceandao.org"

# your infura key
INFURA_KEY = "01SAMPLE9234KEY"

# set to "enabled" if you want to enable wallet address debugging inside app
DEBUGGING = "enabled"


____ staging (prod) is working _____

`
# Choose one of the following environment configuration: local (development), goerli (staging), mainnet (production)
# This will automatically set other required environment variables to point to choosen configuration.

# Development: point to your local ganache chain. Usit it when running Ocean stack locally with Barge.
# NODE_ENV = "development"

# Staging: point to Goerli network.
NODE_ENV = "staging"

# Production: point to Ethereum Mainnet.
# NODE_ENV = "production"

# You can enable or disable rollup livereload to start a server and reload the page on saving changes.
# This is usualy enabled on local and disable on production.
START_RELOAD_SERVER = true

# You can also set your custom environment variables by adding a different value for NODE_ENV then the specified ones above, and manually setting the other variables.

# set one of Ocean subgraphs from the available subgrapsh list from https://github.com/oceanprotocol/ocean-subgraph.
SUBGRAPH_API = "https://v4.subgraph.goerli.oceanprotocol.com/subgraphs/name/oceanprotocol/ocean-subgraph"

# chain where ve contracts are deployed - 8996 is development
VE_SUPPORTED_CHAINID = 8996

# list of suppported chainIDs (used for filtering asset only)
SUPPORTED_CHAIN_IDS = [8996]

# api endpoint to get the pools list
BACKEND_API = "https://test-df-sql.oceandao.org"

# your infura key
INFURA_KEY = "01SAMPLE9234KEY"

# set to "enabled" if you want to enable wallet address debugging inside app
DEBUGGING = "enabled"
`

**Comment by KatunaNorbert** [comment]
Yeah, I forgot to uncomment the Infura key env var

**Comment by EnzoVezzaro** [comment]
I'm a bit confused here, I'm not sure what the end goal is. What I would expect is the following:

1. User clones the repo
2. It follows the instructions
3. when he runs `cp .env.example .env`, he'll get a workable env (in my opinion it should be `development` or `staging` by default). The important this here is that the user doesn't have necessarily to edit the .env file to make the whole project work. 
4. User runs the app in "default" ( `development` or `staging`)

That's why I was proposing something like [this](https://gist.github.com/EnzoVezzaro/8c61129d53582682dc9457e2ad390191). But if we go this route, we should also update our docs to reflect this "default" state of the app.

**Comment by KatunaNorbert** [comment]
That's right @EnzoVezzaro, isn't the app working now if you copy the .env.example to .env?

**Comment by jamiehewitt15** [comment]
> I'm a bit confused here, I'm not sure what the end goal is. What I would expect is the following:
> 
> 1. User clones the repo
> 2. It follows the instructions
> 3. when he runs `cp .env.example .env`, he'll get a workable env (in my opinion it should be `development` or `staging` by default). The important thing here is that the user doesn't have necessarily to edit the .env file to make the whole project work.
> 4. User runs the app in "default" ( `development` or `staging`)
> 
> That's why I was proposing something like [this](https://gist.github.com/EnzoVezzaro/8c61129d53582682dc9457e2ad390191). But if we go this route, we should also update our docs to reflect this "default" state of the app. And in the case of `development`, document the need for barge.

Yeah, I completely agree. This flow isn't working for me - I think the issue is that `SUBGRAPH_API` is commented out

**Comment by jamiehewitt15** [comment]
It's also still not working for me on development. @KatunaNorbert can you share the .env settings you're using for development?

image

**Comment by EnzoVezzaro** [comment]
> I think the issue is that `SUBGRAPH_API` is commented out

on my end too, we need to comment it out to make it work on `staging`.

**Comment by jamiehewitt15** [comment]
@EnzoVezzaro have you managed to get it to work on development?

**Review by jamiehewitt15 — APPROVED** [review]
Managed to get it working. Good work @KatunaNorbert

**Review by EnzoVezzaro — APPROVED** [review]

**Pull request merged**